app/models/assignment.rb

```diff
@@ -1284,13 +1302,14 @@ def self.upload_assignment_list(file_format, assignment_data)
       begin
         map = assignment_data.deep_symbolize_keys
         map[:assignments].map do |row|
-          assignment = self.find_or_create_by(short_identifier: row[:short_identifier],
-                                              assignment_properties: { repository_folder: row[:short_identifier] })
+          assignment = self.find_or_create_by(short_identifier: row[:short_identifier])
           if assignment.new_record?
+            row[:assignment_properties] = AssignmentProperties.new
             row[:submission_rule] = NoLateSubmissionRule.new
             row[:assignment_stat] = AssignmentStat.new
-            row[:token_period] = 1
-            row[:unlimited_tokens] = false
+            row[:assignment_properties][:repository_folder] = row[:short_identifier]
```

Comment:
Similar to the above comment, pass these three attribute values directly to the `AssignmentProperties.new` call above.